### PR TITLE
Adds a setting to make MDCBottomSheets impossible to dismiss.

### PR DESCRIPTION
**Pull request by jmdetloff**
Exported from an internally lgtm'ed CL: cl/192175794

### CODE REVIEW

**Comment by jmdetloff** [comment]
Hmm, have a messed up a step here? Why did this end up including these merge commits?

**Comment by jmdetloff** [comment]
Authored by a googler, ok cla bot

**Comment by jmdetloff** [comment]
cl/192175794

**Comment by romoore** [comment]
Was this through our internal contributing process (using our export tool) or manually exported?

File: components/BottomSheet/src/MDCBottomSheetController.h [hunk 1]
```diff
@@ -45,6 +45,11 @@
  */
 @property(nonatomic, weak, nullable) UIScrollView *trackingScrollView;
 
+/**
+ When set to true, the bottom sheet controller can't be dismissed by tapping outside of sheet area.
```
Comment (jverkoey) [inline components/BottomSheet/src/MDCBottomSheetController.h]: This comment is documenting the opposite of what I think the behavior is.

File: components/BottomSheet/src/MDCBottomSheetController.h [hunk 2]
```diff
@@ -45,6 +45,11 @@
...
+ */
+@property(nonatomic, assign) BOOL dismissOnBackgroundTap;
```
Comment (jverkoey) [inline components/BottomSheet/src/MDCBottomSheetController.h]: As a Bool, this API may read better as "canDismissOnBackgroundTap"

File: components/BottomSheet/src/MDCBottomSheetController.h (hunk 1 above)
Comment (jverkoey) [reply components/BottomSheet/src/MDCBottomSheetController.h]: Looks like the boolean's implementation is the inverse of its name - either change the implementation or the name.

**Review by jverkoey — DISMISSED** [review]
Naming is inverted.

File: components/BottomSheet/src/MDCBottomSheetController.h (hunk 2 above)
Comment (brianjmoore) [reply components/BottomSheet/src/MDCBottomSheetController.h]: This is named to be consistent with `MDCDialogPresentationController`:

https://github.com/material-components/material-components-ios/blob/89f8107445af293a14d916a977b32a86a6fe1d82/components/Dialogs/src/MDCDialogPresentationController.h#L43

**Comment by jmdetloff** [comment]
Merged in requested fixes from original author's cl. @romoore this was generated with our internal export tool

**Comment by romoore** [comment]
@jmdetloff Ok. Looks like we have a bug in the export tool. It seems to be pulling in extra commits.

File: components/BottomSheet/src/MDCBottomSheetController.h [hunk 3]
```diff
@@ -45,6 +45,11 @@
  */
 @property(nonatomic, weak, nullable) UIScrollView *trackingScrollView;
 
+/**
+ When set to false, the bottom sheet controller can't be dismissed by tapping outside of sheet area.
```
Comment (jverkoey) [inline components/BottomSheet/src/MDCBottomSheetController.h:49]: Please document that this is enabled by default.

**Review by jverkoey — APPROVED** [review]
LGTM pending documentation. Feel free to address the docs in a follow-up PR (file a pivotal story to track).

**Pull request merged**